src/vcpkg/vcpkglib.cpp

```diff
@@ -112,11 +112,8 @@ namespace vcpkg
             return; // File already in the new format
         }
 
-        if (!was_tracked)
-        {
-            was_tracked = true;
+        if (was_tracked)
```

Comment:
This is a misunderstanding -- `was_tracked` is a boolean for whether this property has been submitted at least once. It is _not_ a setting for whether metrics are enabled or not.

The way metrics are intended to work in vcpkg is that the apis (`track_property`) are called unconditionally and whether they are actually sent is controlled centrally.

Reply:
OK

If `was_tracked` is `true`,
`LockGuardPtr<Metrics>(g_metrics)->track_property("listfile", "update to new format");` will not be called.
Is it as expected?